src/resmom/linux/alps.c

```diff
@@ -1585,7 +1680,19 @@ node_start(ud_t *d, const XML_Char *el, const XML_Char **atts)
 		return;
 	}
 	/* Reset the array counters. */
-	d->count.segment_array = 0;
+	switch (basilver) {
+	case basil_1_1:
+	case basil_1_2:
+		d->count.segment_array = 0;
+		break;
+	case basil_1_3:
+	case basil_1_4:
+		/* segment_array is reset in socket_start() for these 
+		 * BASIL versions
```

Comment:
We need not have the cases for 1.3 and 1.4. If you decide to keep, just add a full-stop to the comment.